Python/Tests/Utilities.Python/PythonTestExplorer.cs

```diff
@@ -147,19 +149,52 @@ public PythonTestExplorer(VisualStudioApp app, AutomationElement element)
             Assert.IsNotNull(Tests, "Tests list is null");
         }
 
+        public void ClearSearchBar() {
+            if (_searchBarElement != null) {
+                InsertTextUsingUIAutomation(_searchBarElement, "");
+            }
+        }
+
         public AutomationElement WaitForItem(params string[] path) {
             // WaitForItem doesn't work well with offscreen items
-            // so collapse to maximize success (up to a point)
-            // TODO: For increased reliability, we'll have to figure out 
-            // how to make WaitForItem work when item is offscreen.
-            Tests.CollapseAll();
+            // so we use the search bar to filter by function name to 
+            // limit the items on screen and then expand all tree items. 
+            // Currently child items dont always load on expand, so we need to call
+            // it multiple times with delay as a work around.
+            InsertTextUsingUIAutomation(_searchBarElement, path[path.Length - 1]);
+
+            Thread.Sleep(250);
+            for (int i = 0; i < path.Length + 1; i++) {
+                Tests.ExpandAll();
+                Thread.Sleep(250);
+            }
+
             return Tests.WaitForItem(path);
         }
 
+        private void InsertTextUsingUIAutomation(
```

Comment:
no need for this anymore if you use the `AutomationWrapper` helper
i know you have a sleep here, but you shouldn't count on arbitrary sleep statements like this
code that looks for something should always retry instead, and then you don't need to randomly sleep after changing something